tests/unit_tests/client/test_event_bus.py

```diff
@@ -18,7 +22,7 @@ def events(mock_stomp_client: StompClient) -> EventBusClient:
 
 def test_context_manager_connects_and_disconnects(
     events: EventBusClient,
-    mock_stomp_client: Mock,
+    mock_stomp_client: StompClient,
```

Comment:
Honestly I never know what to do with type annotations for mocks. If you annotated with the mocked type you get code completion for its methods but you don't get `assert_called_once` et al. and vice versa.

I tend to annotate based on whether I'm using the real methods or the mock utilities more often but we should probably have a cohesive answer. I've tried using `Mock(spec=ActualType)` before but vscode doesn't seem to pick that up.